tools/rna_tools/reago/reago.xml

```diff
@@ -0,0 +1,140 @@
+<tool id="reago" name="Reago" version="0.1.0">
+    <description> to assemble rRNA</description>
+
+    <requirements>
+        <requirement type="package" version="1.1">infernal</requirement>
+        <requirement type="package" version="1.5.7">genometools</requirement>
+        <requirement type="package" version="1.1">reago</requirement>
+    </requirements>
+
+    <stdio>
+        <exit_code range="1:" />
+    </stdio>
+
+    <command>
+<![CDATA[
+        #set $r1_output=$os.path.splitext($os.path.basename(str($r1_file)))[0]
+        #set $r2_output=$os.path.splitext($os.path.basename(str($r2_file)))[0]
+
+        python $__tool_directory__/format_reago_input_files.py 
+            --r1_input "$r1_file"
+            --r2_input "$r2_file"
+            --r1_output $r1_output
+            --r2_output $r2_output
+
+        ; 
+        python \${REAGO_DIR}/filter_input.py 
+            $r1_output
+            $r2_output
+            .
+            \${REAGO_DIR}/cm/ 
+            $cm_to_use
+            $cpu_nb
+
+        ;
+        python \${REAGO_DIR}/reago.py 
+            "filtered.fasta"
+            .
+            -l $read_length
+            -o $overlap
+            -t $tip_size 
+            -b $path_finding_parameter
+]]>
+    </command>
+
+    <inputs>
+        <param name="r1_file" type="data" format="fasta" label="R1 input sequence 
+            file" help=""/>
+
+        <param name="r2_file" type="data" format="fasta" label="R2 input sequence 
+            file" help=""/>
+
+        <param name="read_length" type="integer" min="1" max="1000" value="101" 
+            label="Read length" help="(-l)"/>
+
+        <param name="overlap" type="float" min="0" max="1" value="0.8" 
+            label="Overlap" help="(-o)"/>
+
+        <param name="tip_size" type="integer" min="0" max="1000" value="30" 
+            label="Tip size" help="(-t)"/>
+
+        <param name="path_finding_parameter" type="integer" min="2" max="11" 
+            value="10" label="Path finding parameter" help="(-b)"/>
+
+        <param name="cm_to_use" type="select" label="Database to use to to identify 16S reads" help="">
+            <option value="b">Bacteria only</option>
+            <option value="a">Archea only</option>
+            <option value="ab">Bacteria and archea</option>
+        </param>
+
+        <param name="cpu_nb" type="integer" min="1" max="10" 
```

Comment:
I would not expose this to the user. But you can use it and give an Galaxy admin the chance to configure it with --cpus "\${GALAXY_SLOTS:-4}" for example.